util/topgen.py

```diff
@@ -1126,17 +1127,7 @@ def main():
         raise SystemExit(sys.exc_info()[1])
 
     # Extract version stamp from file
-    version_stamp = {}
-    if args.version_stamp is not None:
-        try:
-            with open(args.version_stamp, 'rt') as f:
-                log.info("version stamp path: {}", args.version_stamp)
-                for line in f:
-                    k, v = line.strip().split(' ', 1)
-                    version_stamp[k] = v
-                    log.info("{} {}", k, v)
-        except ValueError:
-            raise SystemExit(sys.exc_info()[1])
+    version_stamp = version_file.VersionInformation(args.version_stamp)
```

Comment:
Not for this PR, but at some point, we may want to also accept user-provided versioning info. If others are to use a release from this repo for tools and IP, the opentitan git revision is not necessarily the source for it (if a release is even encapsulated in a git repo at all!).

Reply:
Perhaps this is quibbling on terms, but the user-provided versioning info is encoded in `BUILD_GIT_VERSION`.  That value is the output of `git describe` and communicates the last git tag, the "distance" from the tag (ie: number of commits) and a git object identifier of the current head at build time.

In the case where one is building exactly at the tag, the extra info (distance, object id) are omitted and the value is exactly the name of the tag.

IMHO, the tag name _is_ the user-provided versioning info as users are responsible for naming the tags.  Is there some additional perspective about what we should consider as user-provided versioning info?

Reply:
I think this is a fair point, we do have any mecanism that works outside the git repository.  It seems to me that the purpose of a release could actually include the production of a file that captures the output of the workspace status during the release, and that is then used by bazel instead of running the script. I am not entirely certain this could be achieved without the passing a config file to bazel or releasing a modified BUILD/WORKSPACE file.